src/lib/route.js

```diff
@@ -56,11 +66,20 @@ class Route {
   combine (alternateRoute) {
     const combinedCurve = this.curve.combine(alternateRoute.curve)
     const combinedHops = this._simpleHops()
+    const havePath = {}
+
+    for (let list of [this.paths, alternateRoute.paths]) {
+      for (let path of list) {
+        havePath[path] = true
```

Comment:
This should do the `JSON.stringify` and `Object.keys(havePath).map(JSON.parse)` trick that you do in join